master/buildbot/steps/transfer.py

```diff
@@ -383,15 +388,49 @@ def start(self):
             return self.finished(SKIPPED)
 
         @defer.inlineCallbacks
-        def uploadSources():
-            for source in sources:
-                result = yield self.startUpload(source, masterdest)
-                if result == FAILURE:
-                    defer.returnValue(FAILURE)
-                    return
-            defer.returnValue(SUCCESS)
+        def globSources(sources):
+            dl = defer.DeferredList([
+                self.runGlob(
```

Comment:
You can add another test in this file after this one.
 https://github.com/buildbot/buildbot/blob/python3/master/buildbot/test/integration/test_transfer.py#L48